server/plugin/api.go

```diff
@@ -184,7 +184,7 @@ func (p *Plugin) checkConfigured(next http.Handler) http.Handler {
 		config := p.getConfiguration()
 
 		if err := config.IsValid(); err != nil {
-			http.Error(w, "This plugin is not configured.", http.StatusNotImplemented)
```

Comment:
`UnprocessableEntity` comes from WebDAV and in practice means the request has the correct syntax but the content is invalid in some way.  It is an error that is expected to be corrected by the caller, not the server. 

`NotImplemented` or `InternalServerError` seems like a more suitable error condition for invalid config. I may not be fully understanding the problem, but  I think it *should* trigger alerts to an admin and the server-side condition should be fixed.  

I think `NotImplemented` was not a bad choice, acting much like if the plugin was disabled but with something to differentiate between disabled versus not configured.

If cloud team really wants this, I'll approve.  I would like to caution against hiding errors in obscure HTTP codes.

Reply:
I do agree with Doug here. Although this creates 500s that we cannot solve on our side as we are not responsible for server and plugin configuration, it is a valid error and probably we should keep it this way. The proper path is to ask the admin to fix the configuration on their side.

Reply:
@stylianosrigas By "we" we mean the server and the administrator of the server. So even if I agree that `Not Implemented` might not be the best choice, I think that the error should be a `5xx` if we want to remain semantically accurate, `4xx` generally means that the client is responsible and for sure here this is not the case.  So my 👍🏾 to the comments of @wiggin77 and @stylianosrigas.

Reply:
Thanks for the feedback, folks. You correctly pointed out that an unconfigured plugin is a server error and not a client one.


Closing.